In [1]:
import pandas as pd
# Load the UCI Student Performance mathematics dataset
# The CSV file uses semicolon ; as the separator
# '../' moves up to main folder, then we navigate into data/raw/
file_path = "../data/raw/student-mat.csv"

df = pd.read_csv(file_path, sep=";")
print("Dataset shape:", df.shape)
df.head()

Dataset shape: (395, 33)


,school,sex,age,address,famsize,Pstatus,Medu,Fedu,Mjob,Fjob,...,famrel,freetime,goout,Dalc,Walc,health,absences,G1,G2,G3
0,GP,F,18,U,GT3,A,4,4,at_home,teacher,...,4,3,4,1,1,3,6,5,6,6
1,GP,F,17,U,GT3,T,1,1,at_home,other,...,5,3,3,1,1,3,4,5,5,6
2,GP,F,15,U,LE3,T,1,1,at_home,other,...,4,3,2,2,3,3,10,7,8,10
3,GP,F,15,U,GT3,T,4,2,health,services,...,3,2,2,1,1,5,2,15,14,15
4,GP,F,16,U,GT3,T,3,3,other,other,...,4,3,2,1,2,5,4,6,10,10


In [2]:
summary = pd.DataFrame({
"variable": df.columns,
"dtype": df.dtypes.astype(str).values,
"n_unique": df.nunique().values,
"n_missing": df.isna().sum().values,
})
display(summary)

,variable,dtype,n_unique,n_missing
0,school,object,2,0
1,sex,object,2,0
2,age,int64,8,0
3,address,object,2,0
4,famsize,object,2,0
5,Pstatus,object,2,0
6,Medu,int64,5,0
7,Fedu,int64,5,0
8,Mjob,object,5,0
9,Fjob,object,5,0


# S9: Data Inspection
In this section, we inspect the dataset structure before cleaning and modeling.
We check the dataset shape, column data types, missing-value information, and
summary statistics.

In [3]:
# S9: Standard Inspection Trio
print("1. Dataset shape:")
print(df.shape)
print("\n2. Dataset information:")
df.info()
print("\n3. Summary statistics for all variables:")
display(df.describe(include="all").T)

1. Dataset shape:
(395, 33)

2. Dataset information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      395 non-null    object
 1   sex         395 non-null    object
 2   age         395 non-null    int64 
 3   address     395 non-null    object
 4   famsize     395 non-null    object
 5   Pstatus     395 non-null    object
 6   Medu        395 non-null    int64 
 7   Fedu        395 non-null    int64 
 8   Mjob        395 non-null    object
 9   Fjob        395 non-null    object
 10  reason      395 non-null    object
 11  guardian    395 non-null    object
 12  traveltime  395 non-null    int64 
 13  studytime   395 non-null    int64 
 14  failures    395 non-null    int64 
 15  schoolsup   395 non-null    object
 16  famsup      395 non-null    object
 17  paid        395 non-null    object
 18  activities  395 non-null    object
 1

,count,unique,top,freq,mean,std,min,25%,50%,75%,max
school,395,2,GP,349,NaN,NaN,NaN,NaN,NaN,NaN,NaN
sex,395,2,F,208,NaN,NaN,NaN,NaN,NaN,NaN,NaN
age,395.0,NaN,NaN,NaN,16.696203,1.276043,15.0,16.0,17.0,18.0,22.0
address,395,2,U,307,NaN,NaN,NaN,NaN,NaN,NaN,NaN
famsize,395,2,GT3,281,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Pstatus,395,2,T,354,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Medu,395.0,NaN,NaN,NaN,2.749367,1.094735,0.0,2.0,3.0,4.0,4.0
Fedu,395.0,NaN,NaN,NaN,2.521519,1.088201,0.0,2.0,2.0,3.0,4.0
Mjob,395,5,other,141,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Fjob,395,5,other,217,NaN,NaN,NaN,NaN,NaN,NaN,NaN


## S9 Inspection Summary Note
The dataset inspection shows the number of rows and columns, the variable names,
the data types, and the basic summary statistics. The `df.info()` output helps
identify whether variables are numeric or categorical and whether any columns
contain missing values.
From the `df.describe(include="all").T` output, three important observations are:
1. The dataset contains 395 rows and 33 columns.
2. Some variables are categorical and will need encoding before machine learning.
3. Some numeric variables should be checked for unusual minimum values, maximum
values, or wide ranges.
The next cleaning step is to check missing values, separate numeric and categorical
columns, encode categorical variables, and confirm the target variable for
prediction.